java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -945,7 +942,7 @@ public void setValueLengthSafe(int index, int length) {
    * @param index   position of element to get
    * @return greater than 0 length for non-null element, 0 otherwise
    */
-  public int getValueLength(int index) {
+  public int getValueLength(long index) {
```

Comment:
if the buffer size is a long, the start offset should also be long?

Reply:
This would be a change for LargeBinary/LargeVarChar since the spec for current variable width values only has a 32-bit offset.